src/main/java/org/cloudfoundry/community/servicebroker/s3/service/S3ServiceInstanceBindingService.java

```diff
@@ -15,65 +15,44 @@
  */
 package org.cloudfoundry.community.servicebroker.s3.service;
 
-import java.util.HashMap;
-import java.util.Map;
-
 import org.cloudfoundry.community.servicebroker.exception.ServiceBrokerException;
 import org.cloudfoundry.community.servicebroker.exception.ServiceInstanceBindingExistsException;
 import org.cloudfoundry.community.servicebroker.model.ServiceInstance;
 import org.cloudfoundry.community.servicebroker.model.ServiceInstanceBinding;
+import org.cloudfoundry.community.servicebroker.s3.plan.BasicPlan;
 import org.cloudfoundry.community.servicebroker.service.ServiceInstanceBindingService;
 import org.springframework.beans.factory.annotation.Autowired;
 import org.springframework.stereotype.Service;
 
-import com.amazonaws.services.identitymanagement.model.AccessKey;
-import com.amazonaws.services.identitymanagement.model.User;
-
 /**
  * @author David Ehringer
  */
 @Service
 public class S3ServiceInstanceBindingService implements ServiceInstanceBindingService {
-
-    private final S3 s3;
-    private final Iam iam;
+    private final BasicPlan basicPlan;
```

Comment:
If this is delegating to a `BasicPlan` it seems like you are going to have to exactly copy the `S3ServiceInstanceBindingService` to handle the shared plan concept.  Shouldn't this be some type of "Plan" interface that this service deals with?